spring-integration-file/src/main/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandler.java

```diff
@@ -139,7 +167,22 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.remoteFileTemplate.send(message, this.mode);
+		String path = this.remoteFileTemplate.send(message, this.mode);
+		if (this.chmod != null && isChmodCapable()) {
+			doChmod(this.remoteFileTemplate, path, this.chmod);
```

Comment:
OK. That's fine, too.
How about the `chmod-expression` then, against the `requestMessage`, instead of this static inconvenience.
And maybe with that we even can get rid of `String` setter on  the matter...